docs/targets/sendgrid.md

```diff
@@ -1,159 +1,71 @@
 ```
-
-* Once this, along with the other aformentioned requirements, have been deployed. It is now possible to implement something like the [awskinesis-sinkbinding.yaml](https://github.com/triggermesh/aws-event-sources/blob/master/config/samples/awskinesis-sinkbinding.yaml) example from the [aws-event-sources](https://github.com/triggermesh/aws-event-sources).
-
-* By replacing the section starting at line [#18](https://github.com/triggermesh/aws-event-sources/blob/93df64d5b298cc73acfb93f835646cc1d06429c4/config/samples/awskinesis-sinkbinding.yaml#L18) with:
-
+"Notification from " + <Event Source> + " Knative Event"
 ```
-  sink:
-    ref:
-      apiVersion: targets.triggermesh.io/v1alpha1
-      kind: ConfluentTarget
-      name: triggermesh-confluent
-```
-
-* You can now deploy this .yaml file. This will complete the setup of the sinkbinding!
-
-*****Please note that to deploy the [awskinesis-sinkbinding.yaml](https://github.com/triggermesh/aws-event-sources/blob/master/config/samples/awskinesis-sinkbinding.yaml) example there are other resources that need to be deployed. Please see the [DEVELOPMENT.md](https://github.com/triggermesh/aws-event-sources/blob/master/DEVELOPMENT.md) doc hosted there for full details***** 
-
 
-### Talking to the Sendgrid Target
+The entire cloud event is passed into the email `body` by default.
 
-* Depending on how you want to use the target. You can set up defaults for all the avalible paraments and pass absolutely nothing to have a static message sent to a defined user. Or you can pass in key:value params to dynamically set all of the variables.
-
-An example of a Cloudevent being passed via a Curl command:
-
-```
-curl -v "10.152.183.229" \
-       -X POST \
-       -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
-       -H "Ce-Specversion: 1.0" \
-       -H "Ce-Type: dev.knative.samples.helloworld" \
-       -H "Ce-Source: dev.knative.samples/helloworldsource" \
-       -H "Content-Type: application/json" \
-       -d '{"FromEmail":"Jeff@triggermesh.com.","ToEmail":"jeffthenaef@gmail.com", \
-         "FromName":"SendgridTarget","ToName":"endUser","Message":"hello"}'
-```
+When a **Message** property is present in the payload, its associated value is used to propagate the `body` of the email.
 
-A list of possible keys:
-- message
-- fromname
-- fromemail
-- toname
-- toemail
+[sgSU]:https://signup.sendgrid.com/
+[sg]:https://sendgrid.com/
+[api]:https://sendgrid.com/docs/ui/account-and-settings/api-keys/
 
-**If the "Message" key:value is omited. The body of the email will contain the entire cloud event**
+[ce]: https://cloudevents.io/
+[ce-jsonformat]: https://github.com/cloudevents/spec/blob/v1.0/json-format.md
+[tm-secret]:https://docs.triggermesh.io/guides/secrets/
+[docs]: https://sendgrid.com/docs/
 
+[^s]: *Note*: If there is not a default value specified at the spec for all of the available fields, an event *MUST* contain all of the following, save **Message**, or the Target **will** **fail**
```

Comment:
This target works a bit strangely compared to other targets due to these two modes of operation. It makes it tricky to explain how things work and leaks in multiple places, like the description of fields in `Deploying an Instance of the Target`, which are only relevant for one of the two modes.

This is not a good place to discuss the design of the target, so the only suggestion that comes to my mind is to add a section after `Prerequisites` that explain the two modes in detail, and reference that section from `Event Types` section to avoid turning the description of event types into a dumpster. It also gives more immediate context to the reader and makes the rest of the document clearer.